From 6cd20e2ca35d845193f6c204597293851bbf6cc4 Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Wed, 13 Jul 2016 17:25:20 +0800
Subject: [PATCH 0347/1345] dts: a3700: add regulators to the sata port

commit  0c95107c199f0d194d8f1400f3bb24164aa7084b from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- Add the regulators for the SATA port, 12v regulator is for the motor
  while 5v regulator is for circuit chip, they share a same pin in
  pca9555, when the pin is set output high, both regulators are turned
  on;
- Since armada-3700 achi is a platform ahci sata controller, and the
  platform ahci sata controller only defines one regulator for sata
  target power - "target-supply", so here we set sata's target-supply
  as the 5v regulator.

Change-Id: Ie2bcfdea1945f79c4b0942b8c8ebb279d21ea8c6
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/31159
Reviewed-by: Victor Gu <xigu@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts |   28 ++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index 9a7d63b..4f4b6e52 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -94,6 +94,33 @@
 				enable-active-high;
 			};
 		};
+
+		reg_sata: pwr-sata {
+			compatible = "regulator-fixed";
+			regulator-name = "pwr_en_sata";
+			regulator-min-microvolt = <12000000>;
+			regulator-max-microvolt = <12000000>;
+			enable-active-high;
+			regulator-boot-on;
+			gpio = <&expander0 2 GPIO_ACTIVE_HIGH>;
+		};
+
+		reg_5v_sata: v5-sata {
+			compatible = "regulator-fixed";
+			regulator-name = "v5.0-sata";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			vin-supply = <&reg_sata>;
+		};
+
+		reg_12v_sata: v12-sata {
+			compatible = "regulator-fixed";
+			regulator-name = "v12.0-sata";
+			regulator-min-microvolt = <12000000>;
+			regulator-max-microvolt = <12000000>;
+			vin-supply = <&reg_sata>;
+		};
+
 	};
 };
 
@@ -129,6 +156,7 @@
 
 /* CON3 */
 &sata {
+	target-supply = <&reg_5v_sata>;
 	status = "okay";
 };
 
-- 
1.7.9.5

